### DISCUSSION TITLE
DuckDB

### DISCUSSION THREAD

**Discussion by pere73** [jeremyevans/sequel#2047] [General]
Hello together,

I have been using Sequel for some time now and am super excited. I mainly use SQLite as database. I think that the future will rather belong to the database [duckdb](https://duckdb.org/), so I would like to use this test with Sequel. 

The following does not work at the moment:

```
require 'sequel
DB = Sequel.connect('duckdb://<path/to/folder/>test.ddb')
```

I have been programming in Ruby for some time, but my knowledge is not sufficient to be able to program an adapter (connector) for DuckDB  from scratch. However, I would dare to take my first steps in this direction based on the Sqlite-Adapter. Where can I see how the Sequel-Sqlite-Adapter programmed?

However, there may already be a professional working on a DuckDB adapter, which I would appreciate much more. Regardless, I would be interested to see how such an adapter is built.

**Comment by jeremyevans** [comment]
You should look at Sequel's existing adapters in lib/sequel/adapters.  The best approach to support a new adapter is to copy an existing aadapter, and modify it to support the underlying driver.  I'd recommend copying the trilogy adapter, since it is the newest and smallest.

**Reply by ShreyanJain9** [reply]
Having a go at this - which of these methods and constants are absolutely necessary to define? I'm trying to keep it very minimal.

**Reply by johnknapp** [reply]
Hey @ShreyanJain9, LMK if you need a hand, I'd _definitely_ like to bring DuckDB into my latest roda/sequel project!

**Reply by pere73** [reply]
Sorry, I totally lost track of the request here. First of all, it would be good if we could simply send queries to the database and receive data records back. After that, other functionalities can be added bit by bit. 

I would like to use this opportunity to learn how such a development would work.
